src/sqlfluff/rules/L050.py

```diff
@@ -66,46 +67,45 @@ class Rule_L050(BaseRule):
     """
 
     targets_templated = True
+    needs_raw_stack = False
+    # TRICKY: Tells linter to only call _eval() ONCE, with the root segment
+    recurse_into = False
+    lint_phase = "post"
```

Comment:
Reworked this rule so it is only called once per parse tree, with the root segment.